include/nuttx/fs/fs.h

```diff
@@ -34,6 +34,7 @@
 #include <stdbool.h>
 #include <time.h>
 
+#include <nuttx/mutex.h>
```

Comment:
Here is the source of your build error. You are having circular dependency here. `#include <nuttx/sched.h>` -> `#include <nuttx/fs/fs.h>` -> `#include <nuttx/mutex.h>` -> `#include <nuttx/sched.h>`.

Now we need to think what to do.

Reply:
Yeah... Seems that the easiest way is to remove `#include <nuttx/sched.h>` from `mutex.h`. But let's also avoid usage of `INVALID_PROCESS_ID` in `mutex.h` and use `#define MUTEX_NO_HOLDER         (pid_t)-1`.